src/nimblepkg/packageparser.nim

```diff
@@ -41,16 +42,21 @@ const reservedNames = [
   "LPT9",
 ]
 
-proc newValidationError(msg: string, warnInstalled: bool,
+proc newValidationError(info: PackageInfo, msg: string, warnInstalled: bool,
                         hint: string, warnAll: bool): ref ValidationError =
   result = newException(ValidationError, msg)
   result.warnInstalled = warnInstalled
   result.warnAll = warnAll
   result.hint = hint
+  result.pkgInfo = info
+
+proc raiseNewValidationError(info: PackageInfo, msg: string,
+                             warnInstalled: bool, hint = "", warnAll = false) =
+  raise info.newValidationError(msg, warnInstalled, hint, warnAll)
 
 proc raiseNewValidationError(msg: string, warnInstalled: bool,
-                             hint: string = "", warnAll = false) =
-  raise newValidationError(msg, warnInstalled, hint, warnAll)
+                             hint = "", warnAll = false) =
+  raise PackageInfo().newValidationError(msg, warnInstalled, hint, warnAll)
```

Comment:
This might not look correct - but in fact it is - it's only used this way in ``validatePackageName`` which is only called from ``readPackageInfo`` where it's not covered in a try/except block so it won't make sense (and will require changing more code) to pass PackageInfo() there.
I also could've made a new newValidationError proc without info: PackageInfo, but I wanted to keep my changes as minimal as possible